homeassistant/components/opentherm_gw/climate.py

```diff
@@ -54,8 +59,9 @@ def __init__(self, gw_dev, options):
         """Initialize the device."""
         self._gateway = gw_dev
         self.friendly_name = gw_dev.name
+        options = options or DEFAULT_OPTIONS
         self.floor_temp = options[CONF_FLOOR_TEMP]
-        self.temp_precision = options.get(CONF_PRECISION)
+        self.temp_precision = options[CONF_PRECISION]
```

Comment:
Wouldn't it make more sense to use `options.get(CONF_PRECISION, DEFAULT_PRECISION)` and then do `options = options or {}` above ? That way you can have 1 missing option too.

Reply:
That should not happen. The config flow creates an entry with no options and the options flow provides default values for all options. However, this may indeed become an issue if more options are added in the future. I will make the change.